dlang/source/flatbuffers/flatbufferbuilder.d

```diff
@@ -0,0 +1,392 @@
+                continue;
+            existingVtable = _vtables[i];
+        }
+
+        if (existingVtable != 0)
+        {
+            // Found a match:
+            // Remove the current vtable.
+            _space = _buffer.length - vtableloc;
+            // Point table to existing vtable.
+            _buffer.put!int(_space, cast(int)(existingVtable - vtableloc));
+        }
+        else
+        {
+            // No match:
+            // Add the location of the current vtable to the list of vtables.
+            if (_numVtables == _vtables.length)
+                _vtables.length *= 2;
+            _vtables[_numVtables++] = offset();
+            // Point table to current vtable.
+            _buffer.put!int(_buffer.length - vtableloc, offset() - vtableloc);
+        }
+
+        destroy(_vtable);
+        _vtable = null;
+        return vtableloc;
+    }
+
+    /** This checks a required field has been set in a given table that has
+        just been constructed.
+    */
+    void required(int table, int field)
+    {
+        import std.string;
+
+        auto table_start = _buffer.length - table;
+        auto vtable_start = table_start - _buffer.get!int(table_start);
+        bool ok = _buffer.get!short(vtable_start + field) != 0;
+        // If this fails, the caller will show what field needs to be set.
+        if (!ok)
+            throw new InvalidOperationException(format("FlatBuffers: field %s must be set.",
+                field));
+    }
+    
+    /**
+        Finalize a buffer, pointing to the given `root_table`.
+        Params:
+            rootTable = An offset to be added to the buffer.
+    */
+    void finish(int rootTable)
+    {
+        prep(_minAlign, int.sizeof);
+        addOffset(rootTable);
+    }
+
+    /**
+        Get the ByteBuffer representing the FlatBuffer.
+        Notes: his is typically only called after you call `Finish()`.
+        Returns:
+            Returns the ByteBuffer for this FlatBuffer.
+    */
+    ByteBuffer dataBuffer()
+    {
+        return _buffer;
+    }
+
+    /** 
+        A utility function to copy and return the ByteBuffer data as a `ubyte[]`
+        Retuens:
+            the byte used in FlatBuffer data, it is not copy.
+    */
+    ubyte[] sizedByteArray()
+    {
+        return _buffer.data[_buffer.position .. $];
+    }
+
+    /**
+        Finalize a buffer, pointing to the given `rootTable`.
+        Params:
+            rootTable = An offset to be added to the buffer.
+            fileIdentifier = A FlatBuffer file identifier to be added to the buffer before `root_table`.
+    */
+    void finish(int rootTable, string fileIdentifier)
+    {
+        import std.string;
+
+        prep(_minAlign, int.sizeof + fileIdentifierLength);
+        if (fileIdentifier.length != fileIdentifierLength)
+            throw new ArgumentException(
+                format("FlatBuffers: file identifier must be length %s.", fileIdentifierLength),
```

Comment:
I see you used `std.string.format` here to format the exception message. But in some other file I see you used just std.conv.to and `~` to build the message. Probably same should be done here. Probably no need for format, as it can add to the code size, but this is just an error path so smaller the better.